subprojects/core/src/main/java/org/gradle/internal/progress/DefaultBuildOperationExecutor.java

```diff
@@ -184,4 +191,33 @@ public void failed(Throwable t) {
             failure = t;
         }
     }
+
+    private static class UnmanagedThreadOperation implements Operation {
+
+        private static final AtomicLong COUNTER = new AtomicLong();
+        private final String id;
+        private final String toString;
+
+        private UnmanagedThreadOperation() {
+            long count = COUNTER.getAndIncrement();
+            id = "unmanaged_" + count;
+            toString = "Unmanaged thread operation #" + count + " (" + Thread.currentThread().getName() + ')';
```

Comment:
To do so we would have to hold both `count` and `Thread.currentThread().getName()` because `toString()` might be called from a different thread. And I'm not sure this is important given that this should only be called for plugins doing nasty stuff with their own threads.